app/src/main/java/com/arctouch/codechallenge/api/TmdbApi.java

```diff
@@ -4,36 +4,48 @@
 import com.arctouch.codechallenge.model.Movie;
 import com.arctouch.codechallenge.model.UpcomingMoviesResponse;
 
-import io.reactivex.Observable;
+import okhttp3.OkHttpClient;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.moshi.MoshiConverterFactory;
 import retrofit2.http.GET;
 import retrofit2.http.Path;
 import retrofit2.http.Query;
 
 public interface TmdbApi {
-
-    String URL = "https://api.themoviedb.org/3/";
-    String API_KEY = "1f54bd990f1cdfb230adb312546d765d";
-    String DEFAULT_LANGUAGE = "pt-BR";
-    String DEFAULT_REGION = "BR";
-
     @GET("genre/movie/list")
-    Observable<GenreResponse> genres(
+    Call<GenreResponse> genres(
             @Query("api_key") String apiKey,
             @Query("language") String language
     );
 
     @GET("movie/upcoming")
-    Observable<UpcomingMoviesResponse> upcomingMovies(
+    Call<UpcomingMoviesResponse> upcomingMovies(
             @Query("api_key") String apiKey,
             @Query("language") String language,
             @Query("page") Long page,
             @Query("region") String region
     );
 
+    @GET("search/movie")
+    Call<UpcomingMoviesResponse> search(
+            @Query("api_key") String apiKey,
+            @Query("query") String query,
+            @Query("page") Long page);
+
     @GET("movie/{id}")
-    Observable<Movie> movie(
+    Call<Movie> movie(
```

Comment:
so, any reason why did you not use this API?

Reply:
Unless I missed something...I don't need it. As I'm passing the movie from the list and the Details Activity isn't accessed directly or via deep link. The details provided from the list is enough,  right? :)